spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -333,6 +334,34 @@ public DoughnutChart generateUserStatisticsDoughnutChart(
         }
     }
 
+
+	@RequestMapping(value = "/dashboard/serviceStatsChartCanvas", method = GET, produces = "application/json")
+    @ResponseBody
+    public DoughnutChart generateServiceStatisticsDoughnutChart(
+            @RequestParam("dateFrom") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodStart,
+            @RequestParam("dateUntil") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodEnd) {
+
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        List<StatisticsPerService> userStatsList = metricsAggregationCustomRepository
+                .getStatisticsOfServicesOwnedByLoggedInUserAndByDateRange(serviceApiRepository
+                        .findByOwnerUsername(auth
+                                .getName()), selectedTimePeriodStart, selectedTimePeriodEnd);
+        DoughnutDataset dataset = new DoughnutDataset()
+                .setLabel("Service API calls")
+                .addBackgroundColors(Color.AQUA_MARINE, Color.LIGHT_BLUE, Color.LIGHT_SALMON, Color.LIGHT_BLUE, Color.GRAY)
+                .setBorderWidth(2);
+        userStatsList.forEach(stats -> dataset.addData(stats.getNoOfCalls()));
+        if (userStatsList.size() > 0) {
```

Comment:
I think this could be a sub-function like `createDoughnutChartFromData(...)` . This would reduce the complexity of the super function.

Reply:
Done